### PR TITLE
[tf.data] move map, map_and_batch serialization tests to kernel_tests

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR moves the serialization tests for `map` and `map_and_batch` datasets to the respective files in `kernel_tests` and `experimental/kernel_tests`:

A part of a larger cleanup as discussed in point 4 of https://github.com/tensorflow/tensorflow/pull/46761#issuecomment-770059963

cc: @jsimsa

### CODE REVIEW

File: tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 from absl.testing import parameterized
 import numpy as np
 
+from tensorflow.python.data.experimental.kernel_tests.serialization import dataset_serialization_test_base
```
Comment (jsimsa) [inline tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py:25]: let's move the dataset_serialization_test_base to `tensorflow.python.data.kernel_tests` and rename it to `checkpoint_test_base.py`

File: tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py [hunk 2]
```diff
@@ -399,5 +400,76 @@ def map_fn(x):
       self.evaluate(get_next())
 
 
+class MapAndBatchDatasetSerializationTest(
```
Comment (jsimsa) [inline tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py:403]: Replace "DatasetSerialization" with "Checkpoint", so `MapAndBatchCheckpointTest`

**Review by jsimsa — COMMENTED** [review]
Thank you. Left a couple of comments related to the test base class. It might make sense to address those in a separate PR before we move the individual tests.

File: tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py (hunk 1 above)
Comment (kvignesh1420) [reply tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py:25]: addressed in https://github.com/tensorflow/tensorflow/pull/47318

**Comment by gbaned** [comment]
@kvignesh1420 Can you please check @jsimsa's comments and keep us posted ? Thanks!

**Comment by kvignesh1420** [comment]
@gbaned As per the discussion, the review comments have been addressed in a separate PR https://github.com/tensorflow/tensorflow/pull/47318. Once it gets merged I can continue with this one.

File: tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py (hunk 2 above)
Comment (kvignesh1420) [reply tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py]: addressed in the commit

**Comment by kvignesh1420** [comment]
@jsimsa as https://github.com/tensorflow/tensorflow/pull/47318 was merged, I have made the changes accordingly.

**Comment by jsimsa** [comment]
I suggest we introduce / adjust `shard_count` for the modified test targets so that the modified tests do not take extra long to run.

**Comment by kvignesh1420** [comment]
@jsimsa I tried sharding the datasets and running the tests but couldn't find any major test time reduction. This is probably due to the small size of the datasets being used for checkpointing. However, for reference, I have added sharded dataset checkpointing in `def testNumParallelBatches()` of `class MapAndBatchDatasetCheckpointTest`

**Review by jsimsa — APPROVED** [review]

**Pull request merged**